controllers/binding/snapshotenvironmentbinding_adapter.go

```diff
@@ -99,6 +104,30 @@ func (a *Adapter) EnsureIntegrationTestPipelineForScenarioExists() (reconciler.O
 	return reconciler.ContinueProcessing()
 }
 
+// EnsureEphemeralEnvironmentsCleanedUp will ensure that ephemeral environment(s) associated with the
+// SnapshotEnvironmentBinding are cleaned up.
+func (a *Adapter) EnsureEphemeralEnvironmentsCleanedUp() (reconciler.OperationResult, error) {
+
+	if !gitops.HaveBindingsFailed(a.snapshotEnvironmentBinding) {
+		return reconciler.ContinueProcessing()
+	}
+
+	deploymentTargetClaim, err := a.loader.GetDeploymentTargetClaimForEnvironment(a.client, a.context, a.environment)
+	if err != nil {
+		a.logger.Error(err, "failed to find deploymentTargetClaim defined in environment %s", a.environment.Name)
+		return reconciler.RequeueOnErrorOrStop(err)
+	}
```

Comment:
Totally optional (I'd still approve even without it): but would it be too weird if we move this method call to `controllers/binding/snapshotenvironmentbinding_controller.go` file, right after we are done extracting the environment there, so [somewhere here](https://github.com/redhat-appstudio/integration-service/pull/228/files#diff-5b178fa36b2385712f39d2575244cfca705af946a979640c73ed86f02e3ff0e0R107)?